pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +94,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthProvider specifies custom authentication for the kubernetes cluster.
+	AuthProvider *AuthProviderConfig `json:"auth-provider,omitempty"`
```

Comment:
Do we see a reason why we should only allow one of these?  We currently support wrapping transports, so it seems a little unusual that we wouldn't support wrapping these as well.

I don't feel very strongly, but if someone has a use-case for it now (custom headers for some kind of proxy and gcp maybe?) we should build that so that the serialization stays the same.  If no one can think of a valid reason now, a conversion and deprecated field is possible later.